wechaty-puppet-hostie/puppet_hostie.go

```diff
@@ -0,0 +1,965 @@
+package puppethostie
+
+import (
+  "context"
+  "encoding/json"
+  "errors"
+  "fmt"
+  "github.com/golang/protobuf/ptypes/wrappers"
+  "github.com/gorilla/websocket"
+  pbwechaty "github.com/wechaty/go-grpc/wechaty"
+  file_box "github.com/wechaty/go-wechaty/wechaty-puppet/file-box"
+  option2 "github.com/wechaty/go-wechaty/wechaty-puppet/option"
+  "github.com/wechaty/go-wechaty/wechaty-puppet/schemas"
+  "google.golang.org/grpc"
+  "io"
+  "log"
+)
+
+// ErrNoEndpoint err no endpoint
+var ErrNoEndpoint = errors.New("no endpoint")
+
+var pbEventType2PuppetEventName = map[pbwechaty.EventType]schemas.PuppetEventName{
```

Comment:
Can we import this schema data from wechaty-puppet?

It will be great if it could be auto-generated, or we should maintain it in a center module so that it can be updated in the future to help other modules updated as well.

Reply:
If placed in wechat-puppet, wechat-puppet relies too much on `github.com / wechaty / go-grpc / wechaty`.

Reply:
I believe we should make the import chain rule more clearer: the `wechaty-puppet-hostie` should rely **as much as possible** on the `github.com/wechaty/go-wechaty/wechaty-puppet` and `github.com/wechaty/go-grpc` because they have different responsibilities.

1. `wechaty-puppet-hostie` should implement the interface from `github.com/wechaty/go-wechaty/wechaty-puppet`, and connect those API from the interface, to the grpc service via `github.com/wechaty/go-grpc`
1. `github.com/wechaty/go-wechaty/wechaty-puppet` should only define an interface for all puppets. (A Abstract Class in OOP, a Protocol in ObjC, or Interface in Go(?))
1. `github.com/wechaty/go-grpc` should be auto-generated from the protocol buffer, so that it can provide the Go API interface directly to our puppet provider.

Reply:
I will move this to wechaty-puppet / scheme later.

Reply:
see https://github.com/wechaty/go-wechaty/pull/32/commits/32b3aee3b7d55cb893276126e72a4d67171c50ed

Reply:
It looks good!